src/explore-education-statistics-admin/src/pages/release/data/components/DataFilesReplacementTableRow.tsx

```diff
@@ -60,13 +64,32 @@ export default function DataFilesReplacementTableRow({
     }
   }, [replacementDataFile?.status, toggleFetchPlan, toggleCanCancel]);
 
-  const handleStatusChange = (
-    _file: DataFile,
-    importStatus: DataFileImportStatus,
+  useEffect(() => {
+    onReplacementStatusChange({
```

Comment:
Rather than bodge it in with as we can conditionally update the object like:
```
onReplacementStatusChange({
        ...dataFile,
        ...(dataFile.replacedByDataFile && {
          replacedByDataFile: {
            ...dataFile.replacedByDataFile,
            hasValidReplacementPlan: plan?.valid ,
          },
        }),
      });
      ```